### ISSUE TITLE
Attribution reproduce(Table 4)

### ISSUE THREAD

**Issue by YunSeoHwan** [facebookresearch/audioseal#81]
Hi!

Thank you for releasing the AudioSeal code, and I truly appreciate your recent responses to my issues.

I'm currently working on reproducing the message attribution results. However, I'm seeing accuracy scores that are very close to 1 regardless of the value of N. Since the numbers are significantly higher than those reported in the paper, I initially suspected an error in my code. But because I'm using the official code provided by AudioSeal, I don't think the issue lies there. I also evaluated on the VCTK dataset, but observed similar results.

I have three questions:

1. Could there be an issue with the reproduction code?

2. During attribution evaluation, was it performed without any attacks applied?

3. Since my results are unusually high, could that indicate a potential problem?

Thank you for taking the time to read my questions.

```python

def generate_watermark_audio(
    tensor_batch: torch.Tensor,
    sample_rate: int,
    model: AudioSeal,
    secret_message: torch.Tensor,
    device: str) -> tp.Optional[torch.Tensor]:
    try:
        audios = tensor_batch.to(device)
        msg_batch = secret_message.to(device)
        with torch.no_grad():
            watermarked_audio = model(audios, sample_rate=sample_rate, message=msg_batch, alpha=1)
        return watermarked_audio
    except Exception as e:
        print(f"generate_watermark error: {e}")
        traceback.print_exc()
        return None

def detect_watermark_audio(
    tensor_batch: torch.Tensor,
    sample_rate: int,
    detector: AudioSeal,
    device: str,
    message_threshold_internal: float = 0.50
    ) -> tp.Tuple[tp.Optional[torch.Tensor], tp.Optional[torch.Tensor]]:
    try:
        audio_tensor_batch = tensor_batch.to(device)
        with torch.no_grad():
            result_batch, message_batch = detector.detect_watermark(
                audio_tensor_batch,
                sample_rate=sample_rate,
                message_threshold=message_threshold_internal
            )
        return result_batch, message_batch
    except Exception as e:
        print(f"detect_watermark error: {e}")
        traceback.print_exc()
        return None, None

all_candidate_metrics = []

for cand_idx in tqdm(range(args.num_messages)):
    candidate_message = candidate_messages[cand_idx]
    total_samples_cand = 0
    detected_samples_cand = 0
    correct_detection_cand = 0

    for batch in dataloader:
        audio = batch["audio"].to(args.device)
        sample_rate = int(batch["sample_rate"][0])
        batch_size = audio.shape[0]
        total_samples_cand += batch_size

        # message batch process
        candidate_msg_batch = candidate_message.unsqueeze(0).repeat(batch_size, 1)
        watermarked_audio = generate_watermark_audio(
            audio, sample_rate, generator, candidate_msg_batch, args.device
        )
        if watermarked_audio is None:
            print("Watermark generation failed.")

        confidences, decoded_messages = detect_watermark_audio(
            watermarked_audio, sample_rate, detector, args.device, args.message_threshold
        )
        if confidences is None or decoded_messages is None:
            print("Watermark detection failed.")
        
        # detection threshold -> FPR
        detection_flags = (confidences > args.message_threshold)

        for i in range(batch_size):
            if detection_flags[i]:
                detected_samples_cand += 1
                distance  = (decoded_messages[i] != candidate_message).sum().item()
                if distance == 0:
                    correct_detection_cand += 1
        
    # acc metric
    acc_candidate = correct_detection_cand / total_samples_cand if total_samples_cand > 0 else float('nan')

    all_candidate_metrics.append({
        "candidate_index": cand_idx,
        "total_samples": total_samples_cand,
        "detected_samples": detected_samples_cand,
        "correct_detection": correct_detection_cand,
        "accuracy": acc_candidate
    })
```
<img width="1182" alt="Image" src="https://github.com/user-attachments/assets/ffc3c7d4-8b7e-4a24-8bba-905ae18830b2" />

**Comment by YunSeoHwan** [comment]
Hi pierrefdz,

I re-ran the reproduction using the code you provided. Attribution was successfully measured, but the watermark presence accuracy under the five attacks—highpass, lowpass, white noise, speed, and smoothing—still doesn't fully match the reported results.
Could there be an issue with the [dataset](https://huggingface.co/datasets/facebook/voxpopuli) I’m using?

<img width="802" alt="Image" src="https://github.com/user-attachments/assets/3b58b90e-99cb-42c0-85ed-e5ca7e326541" />

**Comment by goannan** [comment]
Hi, I got the same result with speed attack. did you know the reason now?

![Image](https://github.com/user-attachments/assets/0cc61646-0dd4-4564-95af-cc06b98dc2cc)

**Comment by antoine-tran** [comment]
@YunSeoHwan It looks to me like the rounding error. Could you provide more context on the environment you used ?

**Comment by kingradio** [comment]
@YunSeoHwan Hello, I got the same result with you in highpass, lowpass, white noise, speed, and smoothing(all the acc are 0.5), have you solved this problem?

**Comment by kingradio** [comment]
@antoine-tran Hello, this is my environment and [evaluation scripts](https://justpaste.it/lcx1m). And I used https://hf-mirror.com/facebook/audioseal/resolve/main/detector_base.pth and https://hf-mirror.com/facebook/audioseal/resolve/main/generator_base.pth as the checkpoint. Is there anything wrong? 

### Environment
> NVIDIA-SMI 580.95.05              Driver Version: 580.95.05      CUDA Version: 13.0  NVIDIA GeForce RTX 4090
audioseal                               0.2.0            pypi_0                 pypi
encodec                                 0.1.1            pypi_0                 pypi
ffmpeg                                  8.0.0            gpl_hc3e963e_905       conda-forge
gmp                                     6.3.0            hac33072_2             conda-forge
huggingface-hub                         0.36.2           pypi_0                 pypi
julius                                  0.2.7            pypi_0                 pypi
libgcc                                  15.2.0           h69a1729_7             defaults
numpy                                   1.26.4           pypi_0                 pypi
pip                                     26.0.1           pyhc872135_0           defaults
scikit-learn                            1.7.2            pypi_0                 pypi
scipy                                   1.15.3           pypi_0                 pypi
sdl2                                    2.32.56          h54a6638_0             conda-forge
sdl3                                    3.2.24           h68140b3_0             conda-forge
tokenizers                              0.20.3           pypi_0                 pypi
torch                                   2.1.0+cu118      pypi_0                 pypi
torchaudio                              2.1.0+cu118      pypi_0                 pypi
tqdm                                    4.67.3           pypi_0                 pypi
transformers                            4.46.3           pypi_0                 pypi

### Result
>  none | frame_acc=0.9996 | iou=1.0000 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
      bandpass | frame_acc=0.9999 | iou=1.0000 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
    boost_10.0 | frame_acc=0.9996 | iou=1.0000 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
      duck_0.8 | frame_acc=0.9996 | iou=1.0000 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
  echo_0.5_0.5 | frame_acc=0.9791 | iou=0.9612 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
 highpass_1500 | frame_acc=0.4966 | iou=0.0122 | dec_acc=0.5000 | bit_clean=1.0 | bit_attack=0.46875
  lowpass_5000 | frame_acc=0.9997 | iou=1.0000 | dec_acc=1.0000 | bit_clean=1.0 | bit_attack=1.0
      pink_0.1 | frame_acc=0.9627 | iou=0.9264 | dec_acc=0.9750 | bit_clean=1.0 | bit_attack=1.0
    white_0.05 | frame_acc=0.5412 | iou=0.0830 | dec_acc=0.5000 | bit_clean=1.0 | bit_attack=0.64375
    speed_1.25 | frame_acc=0.5998 | iou=0.0009 | dec_acc=0.5000 | bit_clean=1.0 | bit_attack=0.4625
     smooth_40 | frame_acc=0.5142 | iou=0.0304 | dec_acc=0.5000 | bit_clean=1.0 | bit_attack=0.48125
